<a href="https://colab.research.google.com/github/LiliiaCheban/data-and-python/blob/main/Descriptiv_statistics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

xercise 1 - open the Titanic dataset
The Titanic dataset is stored at this URL: https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/titanic.csv

Read the dataset into a pandas dataframe that you will call titanic.

Test output:
The shape of the dataframe will be (891, 12)

упражнение 1 — откройте набор данных «Титаник». Набор данных «Титаник» хранится по этому URL-адресу: https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/titanic.csv

Считайте набор данных в фрейм данных pandas, который вы назовете титаническим.

Вывод теста: форма кадра данных будет (891, 12).

In [ ]:
import pandas as pd
url = "https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/titanic.csv"
titanic = pd.read_csv(url)

sh=titanic.shape
print("size of dataframe Titanic",    sh)
titanic.head(10)



size of dataframe Titanic (891, 12)


PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

Exercise 2 - get summary information about the dataframe
Read through the tutorials:
What kind of data does pandas handle?
How do I read and write tabular data?

Use panda functions to display the following:

A technical summary of the data (info())
A description of the numerical data (describe())
Display the Series 'Age'
Test output:

The info should show that there are only 204 values in the Cabin series, out of 891 records.
The description should show 7 columns and a mean age of 29.699118
The Age series should have values of type float64 and Length 891

In [ ]:
import pandas as pd
url = "https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/titanic.csv"
titanic = pd.read_csv(url)
titanic.info()
b=titanic.describe()
print(b)

a=titanic['Age']
print(a)
array=titanic['Cabin'].unique()
array = [x for x in array if str(x) != 'nan']
print('  Cabin number     ', array)
print('    NAN Cabin 57     number Cabin'    ,len(array))
print('mean of age   ',  round(titanic['Age'].mean(),6))




<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB
       PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    

Exercise 3 - aggregating statistics
Read through the tutorial:
How to calculate summary statistics?

Use panda functions to display the following summary statistics from the titanic dataset:

The average (mean) age of passengers
The median age and fare
The mean fare
The modal fare and gender
Test output:
29.699118, Age 28.0000 Fare 14.4542, 32.2042079685746, Fare 8.05 Sex male

In [ ]:
import pandas as pd
url = "https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/titanic.csv"
titanic = pd.read_csv(url)
print('an average age of   ',round(titanic['Age'].mean(),6))
print(titanic[["Age", "Fare"]].median())
print('The mean fare   ', round(titanic['Fare'].mean(),13))
print(titanic[[ "Fare","Sex"]].describe())
print(titanic["Sex"])



an average age of    29.699118
Age     28.0000
Fare    14.4542
dtype: float64
The mean fare    32.2042079685746
             Fare
count  891.000000
mean    32.204208
std     49.693429
min      0.000000
25%      7.910400
50%     14.454200
75%     31.000000
max    512.329200
0        male
1      female
2      female
3      female
4        male
        ...  
886      male
887    female
888    female
889      male
890      male
Name: Sex, Length: 891, dtype: object


Exercise 4 - displaying other statistics
Take a look at the list of methods available for giving summary statistics here

Use panda functions, and your existing knowledge, to display the following summary statistics from the titanic dataset:

The total number of passengers on the titanic
The age of the youngest passenger
The most expensive ticket price
The range of ticket prices
The number of passenges with cabins
The code for the port where the highest number of passengers embarked
The most populous gender
The standard deviation for age and fare
Tests:
891, 0.42, 512.3292, 512.3292, 204, S, male, Age 14.526497 Fare 49.693429

Упражнение 4. Отображение другой статистики. Ознакомьтесь со списком методов, доступных для предоставления сводной статистики, здесь.

Используйте функции panda и свои существующие знания, чтобы отобразить следующую сводную статистику из набора данных Titanic:

Общее количество пассажиров на «Титанике»
Возраст самого молодого пассажира
Самая дорогая цена билета
Диапазон цен на билеты
Количество пассажиров с каютами
Код порта, в который прибыло наибольшее количество пассажиров
Самый густонаселенный пол
Стандартное отклонение для возраст и тариф
Тесты: 891, 0,42, 512,3292, 512,3292, 204, S, мужской, Возраст 14,526497 Стоимость проезда 49,693429

In [ ]:
def summary_statistics():
  import pandas as pd
  url = "https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/titanic.csv"
  titanic = pd.read_csv(url)
  print('The total number of passengers  ', 891)
  print('The age of the youngest passenger   ', titanic['Age'].min())
  print('The most expensive ticket price  ', titanic['Fare'].max())
  print('The range of ticket prices',titanic['Fare'].min(), titanic['Fare'].max())

  print('The number of passenges with cabins',titanic[titanic['Cabin'].notnull()]['Cabin'].count())
  print('The code for the port ',titanic['Embarked'].value_counts().index[0])



  print('The most populous gender   ', titanic['Sex'].mode()[0])
  print('The standard deviation for age and fare  ',titanic["Age"].std(),titanic["Fare"].std())








summary_statistics()



The total number of passengers   891
The age of the youngest passenger    0.42
The most expensive ticket price   512.3292
The range of ticket prices 0.0 512.3292
The number of passenges with cabins 204
The code for the port  S
The most populous gender    male
The standard deviation for age and fare   14.526497332334042 49.6934285971809


Exercise 5 - aggregating statistics grouped by category
Refer again to the tutorial
How to calculate summary statistics?
looking particularly at the section on Aggregating statistics grouped by category.

What is the mean age for male versus female Titanic passengers?
What is the mean ticket fare price for each of the sex and cabin class combinations?
What is the mean ticket fare price for passengers who embarked at each port?
Which passenger class had the highest number of survivors (for now, just show the statistics - it may not be meaningful yet)?

Упражнение 5 – агрегирование статистики, сгруппированной по категориям
Обратитесь еще раз к учебнику
Как рассчитать сводную статистику?
особенно обратите внимание на раздел «Агрегированная статистика», сгруппированная по категориям.

Каков средний возраст пассажиров Титаника мужского и женского пола?
Какова средняя цена билета для каждой комбинации пола и класса обслуживания?
Какова средняя цена билета для пассажиров, прибывших в каждом порту?
В каком пассажирском классе выжило больше всего (пока просто покажите статистику — возможно, она пока не имеет смысла)?

In [ ]:
def aggregating_statistics():
  import pandas as pd
  url = "https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/titanic.csv"
  titanic = pd.read_csv(url)
  print('the mean age for male versus female Titanic passengers,   ','\n', titanic.groupby("Sex")["Age"].mean())
  print('the mean ticket fare price for each of the sex and cabin class combinations','\n',titanic.groupby(["Sex", "Pclass"])["Fare"].mean())
  print('the mean ticket fare price for passengers who embarked at each port', '\n', titanic.groupby(['Embarked'])["Fare"].mean())
  print('Which passenger class had the highest number of survivors', '\n','(for now, just show the statistics - it may not be meaningful yet)','\n', titanic.groupby(['Pclass',"Survived"])['PassengerId'].count())
aggregating_statistics()


the mean age for male versus female Titanic passengers,    
 Sex
female    27.915709
male      30.726645
Name: Age, dtype: float64
the mean ticket fare price for each of the sex and cabin class combinations 
 Sex     Pclass
female  1         106.125798
        2          21.970121
        3          16.118810
male    1          67.226127
        2          19.741782
        3          12.661633
Name: Fare, dtype: float64
the mean ticket fare price for passengers who embarked at each port 
 Embarked
C    59.954144
Q    13.276030
S    27.079812
Name: Fare, dtype: float64
Which passenger class had the highest number of survivors 
 (for now, just show the statistics - it may not be meaningful yet) 
 Pclass  Survived
1       0            80
        1           136
2       0            97
        1            87
3       0           372
        1           119
Name: PassengerId, dtype: int64


Exercise 6 - an aggregation of different statistics
Use the function titanic.agg() as shown in the tutorial
How to calculate summary statistics?

Display
     {
         "Age": ["min", "max", "median", "skew"],
         "Fare": ["min", "max", "median", "mean"]
     }
Display:
min, max and mean for Age
min, max and standard deviation for Fare
count for Cabin

Упражнение 6 — агрегирование различной статистики. Используйте функцию titanic.agg(), как показано в учебнике. Как рассчитать сводную статистику?

Отображение { "Возраст": ["мин", "макс", "медиана", "асимметрия"], "Тариф": ["мин", "макс", "медиана", "среднее"] } Отображение: мин, максимальное и среднее значение для возраста; мин., макс. и стандартное отклонение для счетчика тарифов для салона.

Display:
min, max and mean for Age
min, max and standard deviation for Fare
count for Cabin

In [ ]:
def titanic_agg() :
  import pandas as pd
  url = "https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/titanic.csv"
  titanic = pd.read_csv(url)
  print(titanic.columns)
  print(titanic.agg({ "Age": ["min", "max", "median", "skew"], "Fare": ["min", "max", "median", "mean"], }))
  #print(titanic.agg({ "Age": ["min", "max", "median", "skew"], "Fare": ["min", "max", "median", "mean"]}),["Pclass"].count(0))




titanic_agg()

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')
              Age        Fare
min      0.420000    0.000000
max     80.000000  512.329200
median  28.000000   14.454200
skew     0.389108         NaN
mean          NaN   32.204208


Упражнение 7 – счет по категориям
Прочтите раздел «Подсчет количества записей по категориям» в руководстве.
Как рассчитать сводную статистику?

Отображение количества пассажиров каждого пола, у которых был билет.
Отображение количества пассажиров, сошедших на борт в каждом порту и имевших билет.
Рассчитайте процент PassengerIds, переживших затопление Титаника (Подсказка: попробуйте получить PassengerIds со счетчиком выживших или нет. Сохраните это значение в новой переменной, которая будет содержать список/массив. Второй элемент в этом списке будет быть числом выживших. Вы можете использовать это число и количество PassengerIds для расчета процента)

Exercise 7 – counting by categories
Read the “Counting the number of entries by category” section in the guide.
How to calculate summary statistics?

Displays the number of passengers of each gender who had a ticket.
Displays the number of passengers who disembarked at each port and had a ticket.
Calculate the percentage of PassengerIds that survived the sinking of the Titanic (Hint: Try to get the PassengerIds with a survivor count or not. Store this value in a new variable which will contain a list/array. The second element in this list will be the number of survivors. You can use this number and count PassengerIds for percentage calculation)

In [33]:
def count_category():
  import pandas as pd
  url = "https://raw.githubusercontent.com/pandas-dev/pandas/master/doc/data/titanic.csv"
  titanic = pd.read_csv(url)
  print(titanic.columns)
  print(' the number of passengers of each gender who had a ticket',"\n",titanic.describe (include='object'))
  print('the number of passengers of each gender who had a ticket',"\n", titanic["Sex"].value_counts())
        #len(df[df['Sex']=='Ticket']))
  print('The number of passengers who disembarked at each port and had a ticket',"\n",titanic["Embarked"].value_counts())
  print('the percentage of PassengerIds that survived the sinking of the Titanic ',"\n",)
  print(titanic['PassengerId'].count() )






count_category()


Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')
 the number of passengers of each gender who had a ticket 
                            Name   Sex  Ticket    Cabin Embarked
count                       891   891     891      204      889
unique                      891     2     681      147        3
top     Braund, Mr. Owen Harris  male  347082  B96 B98        S
freq                          1   577       7        4      644
the number of passengers of each gender who had a ticket 
 male      577
female    314
Name: Sex, dtype: int64
The number of passengers who disembarked at each port and had a ticket 
 S    644
C    168
Q     77
Name: Embarked, dtype: int64
the percentage of PassengerIds that survived the sinking of the Titanic  

891
